jbehave-core/src/main/java/org/jbehave/core/steps/StepFinder.java

```diff
@@ -121,28 +124,56 @@ public StepFinder(PrioritisingStrategy prioritisingStrategy) {
         return candidateSteps.stream()
             .map(CandidateSteps::listCandidates)
             .flatMap(List::stream)
-            .collect(Collectors.groupingBy(c -> c.getStepType() + " " + c.getPatternAsString(),
-                    Collectors.mapping(Function.identity(), Collectors.toList()))).entrySet().stream()
+            .collect(stepCandidateCollector())
+            .entrySet()
+            .stream()
             .map(e -> {
                 List<StepCandidate> candidates = e.getValue();
                 if (candidates.size() == 1) {
+                    if(isAnnotationPresent(Conditional.class, candidates)) {
```

Comment:
`isAnnotationPresent(Conditional.class, candidates)` can be extracted to a local variable